internal/services/storage/custompollers/data_plane_queues_availability_poller.go

```diff
@@ -35,6 +36,10 @@ func NewDataPlaneQueuesAvailabilityPoller(ctx context.Context, client *storageCl
 
 func (d *DataPlaneQueuesAvailabilityPoller) Poll(ctx context.Context) (*pollers.PollResult, error) {
 	resp, err := d.client.GetServiceProperties(ctx)
+	var e pollers.PollingDroppedConnectionError
+	if errors.As(err, &e) {
+		return nil, err
+	}
```

Comment:
The other two availability pollers return the `PollingDroppedConnectionError` based on whether `resp.HttpResponse == nil`. Can we not use the same pattern for checking here?

Reply:
Unfortunately the API call `GetServiceProperties` does not return the same struct, and does not contain `HttpResponse`, it instead returns the details and an error type. We check that error type here via `errors.As()` to take the appropriate action.